src/main/java/roomescape/reservation/repository/JdbcReservationRepository.java

```diff
@@ -152,7 +170,43 @@ public List<Reservation> findAll() {
                 ON r.time_id = rt.id
                 INNER JOIN themes AS th
                 ON r.theme_id = th.id
+                INNER JOIN members AS m
+                ON r.member_id = m.id
                 """;
         return jdbcTemplate.query(sql, RESERVATION_ROW_MAPPER);
     }
+
+    @Override
+    public List<Reservation> findByInFromTo(final Long themeId, final Long memberId, final LocalDate dateFrom,
+                                            final LocalDate dateTo) {
+        final String sql = """
+                SELECT 
+                    r.id AS id,
+                    r.date AS date,
+                    m.id AS member_id,
+                    m.name AS member_name,
+                    m.email AS email,
+                    m.password AS password,
+                    m.role AS role,
+                    rt.id AS time_id,
+                    rt.start_at AS start_at,
+                    th.id AS theme_id,
+                    th.name AS theme_name,
+                    th.description AS description,
+                    th.thumbnail AS thumbnail
+                FROM reservations AS r
+                INNER JOIN reservation_times AS rt
+                ON r.time_id = rt.id
+                INNER JOIN themes AS th
+                ON r.theme_id = th.id
+                INNER JOIN members AS m
+                ON r.member_id = m.id
+                WHERE ((? IS NULL OR ? IS NULL) OR (r.date BETWEEN ? AND ?))
+                AND (th.id = ? OR ? IS NULL)
```

Comment:
```suggestion
                AND (th.id = :themeId OR :themeId IS NULL)
```
위와 같이 바인딩되는 파라미터명을 명시해보는건 어떨까요? 중복으로 바인딩하는 걸 줄일 수 있고, 알아보기도 쉬워질 것 같네요 :)

Reply:
헉,, 해당 부분에 대한 것은 이번에 처음 알았어요! NamedParameterJdbcTemplate를 사용하는 방법이군요!,,
해당 부분에 대해서 위치 기반 바인딩이 아닌 이름 기반 바인딩을 통해 직관성을 높일 수 있을 것 같다는 장점이 있을 것 같아요.

단순 쿼리를 사용해서 나타낼 수 있는 간단한 쿼리같은 경우는 JdbcTemplate를 사용해도 알아보기 쉬울 것 같은데,
현재 해당 부분과 같이 바인딩을 해야 하는 곳이 8군데 정도로 많은 경우에는 이름 기반으로 바인딩 하는 것이 좋을 것 같아 수정했습니다!